### DISCUSSION TITLE
Snippets not showing in typescript files

### DISCUSSION THREAD

**Discussion by blancpain** [LazyVim/LazyVim#3344] [Q&A] [answered]
Hi all,

Im using the default snippet setup with nvim-snippets with friendly-snippets enabled. Nothing fancy in my cmp setup, have just added super-tab support (following the recipe in the docs...). I am noticing that snippets just don't show up at all in .ts/.tsx files, while they do show in js, lua etc files. 

Has anyone also noticed this? Any fixes? Haven't had an issue before with luasnip.

**Comment by chmouel** [comment]
make sure you don't have json errors in that files that is not supported by the default json parser,  i usually run them via jq . to check those, compared to LuaSnip the builtin snippets are much more pendantic about json errors

**Comment by blancpain** [comment]
Can you please elaborate on this? I don't think it's related to json errors. I've tested it in multiple ts projects and snippets aren't showing up in the cmp menu. Also it only happens for TS/TSX files so not sure how json errors are relevant as it works for all other filetypes.

**Comment by abereghici** [comment]
I'm facing the same problem. Looks like the snippets are loaded only in `.js` files. `.ts, .tsx, .jsx` don't work.

**Comment by folke** [comment]
I just pushed a change that inludes javascript for more filetypes:

```lua
              extended_filetypes = {
                typescript = { "javascript" },
                typescriptreact = { "javascript" },
                javascriptreact = { "javascript" },
              },
```

**Reply by VesperQuartz** [reply]
mine doesn't work either for react(tsx|jsx) file. same issue the original poster is complaining about. it works for base js|ts file

**Comment by blancpain** [comment]
Awesome, thanks folke!

Snippets seem to be showing up now. I noticed another issue though when I try to run a snippet like "tsrfce" to initiate a typescript react file I get a "notify.error vim.snippet Failed to parse snippet" - is there a workaround for that?

**Reply by dpetka2001** [reply]
Maybe this is a nested snippet? Native `vim.snippet` doesn't have support for nested snippets. Your best bet is to keep using LuaSnip for these.

**Reply by folke** [reply]
I just checked and that snippet is actually invalid since it has multiple tabstops with the same id

**Reply by folke** [reply]
See here https://github.com/rafamadriz/friendly-snippets/blob/d0610077b6129cf9f7f78afbe3a1425d60f6e2f1/snippets/javascript/react-es7.json#L55

**Reply by folke** [reply]
And I just fixed that in LazyVim, so those snippets will work fine now.

**Reply by blancpain** [reply]
Sweet, thanks for the quick turnaround! 

The snippets work now but noticed a few other issues - we now get a warning - "notify.warn vim snippet Failed to parse snippet, but was able to fix it automatically." The snippet is parsed but the "TM_FILENAME_BASE" bit seems to not work as the component name is left blank with the cursor waiting for the user input. Even when you input the name the same is not updated in the export default part (if we look at the "tsrfce" snippet). In any case the name should automatically be taken from the filename, the default behaviour. 

EDIT: The component naming bit now seems to be broken for jsx snippets as well (e.g. "rfce").

**Reply by folke** [reply, marked as answer]
Yes, this will ultimately be fixed in Neovim. I'm just making sure that snippets at least work partially when they cant be parsed correctly.

**Reply by blancpain** [reply]
Thanks folke for all the efforts! Actually seems to be working nicely already (haven't tested with custom snippets but the default ones at least seem to be OK)!

**Comment by folke** [comment]
This PR will add full support for friendly snippets: https://github.com/garymjr/nvim-snippets/pull/31

**Reply by folke** [reply]
Merged!
